dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/SchedulerStart.java

```diff
@@ -130,59 +157,98 @@ public void execute( )
+        if ( verifyNoJobExist( DEFAULT_VALIDATION_RESULTS_NOTIFICATION, jobConfigurations ) )
         {
-            JobConfiguration validationResultNotification = new JobConfiguration( DEFAULT_VALIDATION_RESULTS_NOTIFICATION,
-                VALIDATION_RESULTS_NOTIFICATION, CRON_DAILY_7AM, null, false, true );
+            JobConfiguration validationResultNotification = new JobConfiguration(
+                DEFAULT_VALIDATION_RESULTS_NOTIFICATION, VALIDATION_RESULTS_NOTIFICATION, CRON_DAILY_7AM, null, false,
+                true );
             addAndScheduleJob( validationResultNotification );
         }
 
-        if ( addDefaultJob( DEFAULT_CREDENTIALS_EXPIRY_ALERT, jobConfigurations ) )
+        if ( verifyNoJobExist( DEFAULT_CREDENTIALS_EXPIRY_ALERT, jobConfigurations ) )
         {
             JobConfiguration credentialsExpiryAlert = new JobConfiguration( DEFAULT_CREDENTIALS_EXPIRY_ALERT,
                 CREDENTIALS_EXPIRY_ALERT, CRON_DAILY_2AM, null, false, true );
             addAndScheduleJob( credentialsExpiryAlert );
         }
 
-        if ( addDefaultJob( DEFAULT_DATA_SET_NOTIFICATION, jobConfigurations ) )
+        if ( verifyNoJobExist( DEFAULT_DATA_SET_NOTIFICATION, jobConfigurations ) )
         {
             JobConfiguration dataSetNotification = new JobConfiguration( DEFAULT_DATA_SET_NOTIFICATION,
                 DATA_SET_NOTIFICATION, CRON_DAILY_2AM, null, false, true );
             addAndScheduleJob( dataSetNotification );
         }
 
-        if ( addDefaultJob( DEFAULT_REMOVE_EXPIRED_RESERVED_VALUES, jobConfigurations ) )
+        if ( verifyNoJobExist( DEFAULT_REMOVE_EXPIRED_RESERVED_VALUES, jobConfigurations ) )
         {
             JobConfiguration removeExpiredReservedValues = new JobConfiguration( DEFAULT_REMOVE_EXPIRED_RESERVED_VALUES,
                 REMOVE_EXPIRED_RESERVED_VALUES, CRON_HOURLY, null, false, true );
             addAndScheduleJob( removeExpiredReservedValues );
         }
+
+        if ( verifyNoJobExist( DEFAULT_LEADER_ELECTION, jobConfigurations ) && "true".equalsIgnoreCase( redisEnabled ) )
+        {
+            JobConfiguration leaderElectionJobConfiguration = new JobConfiguration( DEFAULT_LEADER_ELECTION,
+                LEADER_ELECTION, String.format( LEADER_JOB_CRON_FORMAT, leaderElectionTime ), null, false, true );
+            leaderElectionJobConfiguration.setLeaderOnlyJob( false );
+            addAndScheduleJob( leaderElectionJobConfiguration );
+        }
+        else
+        {
+            checkLeaderElectionJobConfiguration( jobConfigurations );
+        }
+        
+    }
+
+    private void checkLeaderElectionJobConfiguration( List<JobConfiguration> jobConfigurations )
+    {
+        JobConfiguration leaderElectionJobConfiguration = jobConfigurations.stream()
+            .filter( jobConfiguration -> jobConfiguration.getName().equals( DEFAULT_LEADER_ELECTION ) ).findFirst()
+            .get();
+        if ( !leaderElectionJobConfiguration.getCronExpression()
```

Comment:
Maybe just set the cron anyways? I guess it is equally fast to just set it. Unless it is configurable to some extent?

Reply:
Got it.